examples/crm/src/providers/fakerest/authProvider.ts

```diff
@@ -50,19 +51,21 @@ export const authProvider: AuthProvider = {
         localStorage.getItem(USER_STORAGE_KEY)
             ? Promise.resolve()
             : Promise.reject(),
-    getPermissions: async () => {
+    canAccess: async ({ signal, ...params }) => {
+        // Get the current user
         const userItem = localStorage.getItem(USER_STORAGE_KEY);
         const localUser = userItem ? (JSON.parse(userItem) as Sale) : null;
-        if (!localUser) {
-            return Promise.reject('user is not logged in');
-        }
+        if (!localUser) return false;
 
-        // We fetch permissions from server to avoid local storage tampering
-        const user = await dataProvider.getOne<Sale>('sales', {
+        // Get the matching sale
+        const { data } = await dataProvider.getOne<Sale>('sales', {
```

Comment:
Correct me if I'm wrong, but I believe permissions were cached, whereas the `canAccess` method is called each time we need it. This would result in many calls to `dataProvider.getOne`.
Can't we instead call that at login, and store the `role` in memory or in localStorage?